### DISCUSSION TITLE
vim.api.nvim_set_keymap not working inside a config function

### DISCUSSION THREAD

**Discussion by craigmac** [wbthomason/packer.nvim#758] [Q&A]
Is it possible to use something like:

```lua
use { 'my/plugin', config = function() vim.api.nvim_set_keymap('n', '<Leader>w', '<Cmd>write<CR>', {}) end }
```

I can set this binding to work if I set it OUTSIDE of the config function, but when put inside the config function it doesn't bind it. Am I missing something?

**Comment by iabdelkareem** [comment]
I have the same issue. I tried to set telescope key bindings via the config function but it doesn't work. Not sure what's wrong.

```lua
packer = require("packer")
use = packer.use

use { 'nvim-telescope/telescope.nvim', requires = { {'nvim-lua/plenary.nvim'}  }, config = function()

        vim.api.nvim_set_keymap("n", "<leader>ff", ":Telescope find_files<cr>", {noremap = true})

        vim.api.nvim_set_keymap("n", "<leader>fg", ":Telescope live_grep<cr>", {noremap = true})

        vim.api.nvim_set_keymap("n", "<leader>fb", ":Telescope buffers<cr>", {noremap = true})

        vim.api.nvim_set_keymap("n", "<leader>fh", ":Telescope help_tags<cr>", {noremap = true})
end
}
```

**Comment by sdondley** [comment]
Place the configuration settings into a lua file in the `lua` directory and write the function to require the file. Easiest and cleanest method. Also saves you from doing "PackerCompile" every time you change your config.

Example

```
 use { 'some-plugin',  config = function() require'the_lua_file_name_here' end }
```

**Reply by iabdelkareem** [reply]
Worked for me thx
